XCode/Sources/TlsSession.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  HttpRouter.swift
+//  Swifter
+//
+//  Copyright (c) 2014-2016 Damian Kołakowski. All rights reserved.
+//
+
+import Foundation
+
+#if !os(Linux)
+private func ensureNoErr(_ status: OSStatus) throws {
+    guard status == noErr else {
+        throw Errno.sslError(from: status)
+    }
+}
+
+public enum TLS {
+    /// Imports .p12 certificate file constructing structure to be used in TLS session.
+    ///
+    /// See [SecPKCS12Import](https://developer.apple.com/documentation/security/1396915-secpkcs12import).
+    /// Apple docs contain a misleading information that it does not import items to Keychain even though
+    /// it does.
+    ///
+    /// - Parameter _data: .p12 certificate file content
+    /// - Parameter password: password used when importing certificate
+    public static func loadP12Certificate(_ _data: Data, _ password: String) throws -> CFArray {
+        let data = _data as NSData
+        let options = [kSecImportExportPassphrase: password]
+        var items: CFArray!
+        try ensureNoErr(SecPKCS12Import(data, options as NSDictionary, &items))
```

Comment:
`SecPKCS12Import(_:_:_:)` is expecting a `UnsafeMutablePointer<CFArray?>`What are differences here between pass a `UnsafeMutablePointer<CFArray?>` and raw `CFArray?`. Wouldn't be more type-safe to pass a type like this:

```swift
try ensureNoErr(withUnsafeMutablePointer(to: &items) { SecPKCS12Import(data, options as CFDictionary, $0) })
```

Reply:
Not fully sure what `&items` generates but I assume it's something like a pointer to `CFArray`. Used https://developer.apple.com/documentation/security/certificate_key_and_trust_services/identities/importing_an_identity as an example.